File: pa-manager-web/src/main/java/com/xiao/sellergoods/controller/BrandController.java
package com.xiao.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.github.pagehelper.PageInfo;
import com.xiao.entity.PageResult;
import com.xiao.entity.Result;
import com.xiao.pojo.TbBrand;
import com.xiao.sellergoods.service.BrandService;
import com.xiao.utils.UsualResult;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * Created by travelround on 2018/12/6.
 */
//1.使用@Controller 注解，在对应的方法上，视图解析器可以解析return 的jsp,html页面，并且跳转到相应页面,若返回json等内容到页面，则需要加@ResponseBody注解
//2.@RestController注解，相当于@Controller+@ResponseBody两个注解的结合，返回json数据不需要在方法前面加@ResponseBody注解了，但使用@RestController这个注解，就不能返回jsp,html页面，视图解析器无法解析jsp,html页面
@RestController
@RequestMapping("/brand")
public class BrandController {
    //日志
    protected final Log logger = LogFactory.getLog(this.getClass());

    // @reference也是注入，但是一般用来注入分布式的远程服务对象，需要配合dubbo配置使用
    @Reference
    private BrandService brandService;
    /**
     * 返回全部列表
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        List<TbBrand> list = brandService.findAll();
        return list;
    }

    @RequestMapping("/findPage")
    public PageResult<TbBrand> findPage(
            @RequestParam(defaultValue = "1") Integer page,
            @RequestParam(defaultValue = "10") Integer rows){
        PageInfo<TbBrand> result = brandService.findPage(page, rows);
        return new PageResult<>(result.getTotal(), result.getList());
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){
        int i;
        try{
            i = brandService.add(tbBrand);
        }catch (Exception e){
            e.printStackTrace();
            return UsualResult.fail;
        }
        return UsualResult.check(i);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        int i;
        try{
            i = brandService.update(tbBrand);
        }catch (Exception e){
            e.printStackTrace();
            return UsualResult.fail;
        }
        return UsualResult.check(i);
    }

    @RequestMapping("/delete")
    public Result delete(@RequestParam(name = "ids") List<Long> ids){
        int i;
        try{
            i = brandService.delete(ids);
        }catch (Exception e){
            e.printStackTrace();
            return UsualResult.fail;
        }
        return UsualResult.check(i);
    }

    @RequestMapping("/search")
    public PageResult<TbBrand> search(
            @RequestParam(defaultValue = "1") Integer page,
            @RequestParam(defaultValue = "10") Integer rows,
            @RequestBody TbBrand tbBrand){
        PageInfo<TbBrand> info = brandService.search(page,rows,tbBrand);
        return new PageResult<>(info.getTotal(), info.getList());
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}

File: pa-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController' ,function($scope,$controller,brandService){

    $controller('bController', {$scope: $scope});//继承

    //删除方法
    $scope.dele = function(){
        $scope.pageObj.dele(brandService.dele,$scope.reloadList());
    };
    //添加或删除选中的数据的id
    $scope.updateSelection = function ($event,id){
        $scope.pageObj.updateSelection($event,id);
    };

    //查找所有数据
    $scope.findAll=function(){
        brandService.findAll().success(
            function(response){
                $scope.pageObj.list = response;
            }
        );
    };

    //缩短调用reloadList方法
    $scope.reloadList = function(){
        $scope.pageObj.reloadList(brandService.search);
    }


    //根据页数和每页多少条查找数据
    $scope.findPage = function (page, rows) {
        brandService.findPage(page,rows).success(
            function (response) {
                $scope.pageObj.list = response.rows;
                $scope.pageObj.conf.totalItems = response.total;//更新总记录数
            }
        );
    };

    //保存数据
    $scope.save = function () {
        $scope.pageObj.save(brandService.add,brandService.update,$scope.pageObj.entity.id);
    };

    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (responce) {
            $scope.pageObj.entity = responce;
        })
    };
});